pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/AggregationGroupByOperatorService.java

```diff
@@ -243,9 +243,19 @@ public AggregationGroupByOperatorService(List<AggregationInfo> aggregationInfos,
     return aggregationResults;
   }
 
-  private Serializable formatValue(Serializable value) {
-    return (value instanceof Float || value instanceof Double) ? String.format(Locale.US, "%1.5f", value)
-        : value.toString();
+  /**
+   * Format the {@link Serializable} value into {@link String}.
+   * <p>For {@link Double} value, format it into the form #####.#####.
+   *
+   * @param value value to be formatted.
+   * @return formatted value.
+   */
+  public static String formatValue(Serializable value) {
+    if (value instanceof Double) {
+      return String.format(Locale.US, "%1.5f", (Double) value);
```

Comment:
do we need the explicit cast for double. I think there is better format pattern to not add a decimal point where the result does not have any decimal. Something like 1.# or something.

why not check for instanceof Number?